server/src/test/java/org/elasticsearch/common/joda/JavaJodaTimeDuellingTests.java

```diff
@@ -33,10 +34,55 @@
 import java.util.Locale;
 
 import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.is;
 
 public class JavaJodaTimeDuellingTests extends ESTestCase {
 
+    public void testIncompatiblePatterns() {
+        // in joda 'y' means year, this is changed to 'u' in java.time. difference is in before era yeaers
+        assertSameMillis("-0001-01-01", "yyyy-MM-dd", "8uuuu-MM-dd");
+        assertSameMillis("-1", "y", "8u");
+
+        // year-of-era in joda becomes 'y' in java.time
+        assertSameMillis("2019-01-01", "YYYY-MM-dd", "8yyyy-MM-dd");
+
+
+        //in joda 'Z' was able to parse 'Z' zulu but in java it fails. You have to use 'X' to do that.
+        //Caused by: java.time.format.DateTimeParseException: Text '2019-01-01T01:01:01.001Z' could not be parsed at index 23
+        //assertSameMillis("2019-01-01T01:01:01.001Z","YYYY-MM-dd'T'HH:mm:ss.SSSZ","8yyyy-MM-dd'T'HH:mm:ss.SSSZ");
```

Comment:
I think these lines might be left over from something? Otherwise I'm not sure why they're here.

Reply:
This whole test case `testIncompatiblePatterns` was meant to use examples of patterns that are incompatible between joda-java and show how to fix them.
For 'Z' we should go for 'X' so I will just remove this comment.